### ISSUE TITLE
Использование неинициализированной переменной

### ISSUE THREAD

**Issue by Stepa86** [1c-syntax/bsl-language-server#111]
```bsl
Процедура СделатьЧтоТоПолезное()

    Если УсловиеВыполняется() Тогда

        полезнаяПеременная = 1;

    Иначе

        СделатьЧтоТоБесполезное(полезнаяПеременная);

    КонецЕсли;

КонецПроцедуры
```

**Comment by zeegin** [comment]
Заложенное поведение платформы: Все неинициализированные переменные равны `Неопределено`.

Часто на это поведение закладываются, например:
```bsl
Процедура СделатьЧтоТоПолезное()

    Если УсловиеВыполняется() Тогда
        полезнаяПеременная = 1;
    КонецЕсли;
    СделатьЧтоТоБесполезное(полезнаяПеременная);

КонецПроцедуры
```

и это будет корректно работающий код, да

В данном случае писать

Часто на это поведение закладываются, например:
```bsl
Процедура СделатьЧтоТоПолезное()

    полезнаяПеременная = Неопределено;
    Если УсловиеВыполняется() Тогда
        полезнаяПеременная = 1;
    КонецЕсли;
    СделатьЧтоТоБесполезное(полезнаяПеременная);

КонецПроцедуры
```
Первую строчку платформа добавит сама.

**Comment by Stepa86** [comment]
Использование того, что платформа сама инициализирует переменную, выглядит как жосткий финт ушами. Код все таки должен быть простым и понятным, без хитростей и необходимости напрягать мозг. Выбирая между вариантами
1) Что то не учли и переменная прилетела с неопределено
2) Это запланированное поведение, но чтоб сонар не ругался приходится явно проинициализировать переменную
я выберу 2

**Comment by zeegin** [comment]
Согласен. Людей, которые ошибутся, теоретически больше, чем тех, кто финтует.
Тем не менее прошу сделать не как критикал, а как предупреждение.

**Comment by Stepa86** [comment]
@zeegin В сонаре есть тип `BUG`, `VULNERABILITY`, `CODE_SMELL` и важность `BLOCKER`, `CRITICAL`, `MAJOR`, `MINOR`, `INFO`. Ты за какое сочетание?

**Comment by asosnoviy** [comment]
https://its.1c.ru/db/v8std#content:494:hdoc
